The default path for GNU programs in Solaris is /usr/gnu/bin which is different
from what hg-git assumes. Change via test environment is not enough because the
test runner is doing some additional transformations with PATH.

Also, some glob results are slightly different on Solaris.

This is a Solaris specific change not intended for upstream.

--- hg-git-0.8.12/tests/run-tests.py
+++ hg-git-0.8.12/tests/run-tests.py
@@ -2501,7 +2501,7 @@
             sepb = _bytespath(os.pathsep)
         else:
             sepb = os.pathsep
-        path = [self._bindir, runtestdir] + osenvironb[b"PATH"].split(sepb)
+        path = ['/usr/gnu/bin', self._bindir, runtestdir] + osenvironb[b"PATH"].split(sepb)
         if os.path.islink(__file__):
             # test helper will likely be at the end of the symlink
             realfile = os.path.realpath(fileb)
--- hg-git-0.8.12/tests/test-git-clone.t
+++ hg-git-0.8.12/tests/test-git-clone.t
@@ -51,8 +51,8 @@
   abort: potentially unsafe hostname: '-oProxyCommand=rm${IFS}nonexistent'
   [255]
   $ hg clone 'git+ssh://fakehost|rm${IFS}nonexistent/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7crm%24%7bifs%7dnonexistent* (glob)
+  ssh: * fakehost%7Crm%24%7BIFS%7Dnonexistent* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
   $ hg clone 'git+ssh://fakehost%7Crm${IFS}nonexistent/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7crm%24%7bifs%7dnonexistent* (glob)
+  ssh: * fakehost%7Crm%24%7BIFS%7Dnonexistent* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
--- hg-git-0.8.12/tests/test-pull.t
+++ hg-git-0.8.12/tests/test-pull.t
@@ -366,8 +366,8 @@
   abort: potentially unsafe hostname: '-oProxyCommand=rm nonexistent'
   [255]
   $ hg pull 'git+ssh://fakehost|shellcommand/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7cshellcommand* (glob)
+  ssh: * fakehost%7Cshellcommand* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
   $ hg pull 'git+ssh://fakehost%7Cshellcommand/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7cshellcommand* (glob)
+  ssh: * fakehost%7Cshellcommand* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
--- hg-git-0.8.12/tests/test-push.t
+++ hg-git-0.8.12/tests/test-push.t
@@ -211,8 +211,8 @@
   abort: potentially unsafe hostname: '-oProxyCommand=rm nonexistent'
   [255]
   $ hg push 'git+ssh://fakehost|rm%20nonexistent/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7crm%20nonexistent* (glob)
+  ssh: * fakehost%7Crm%20nonexistent* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
   $ hg push 'git+ssh://fakehost%7Crm%20nonexistent/path' 2>&1 >/dev/null | grep -v ^devel-warn:
-  ssh: * fakehost%7crm%20nonexistent* (glob)
+  ssh: * fakehost%7Crm%20nonexistent* (glob)
   abort: git remote error: The remote server unexpectedly closed the connection.
